File: 0007-reverse-integer/0007-reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        """
        Reverse the digits of a 32-bit signed integer.
        Returns 0 if the reversed integer overflows.
        """
        # Extract sign and work with absolute value
        sign = -1 if x < 0 else 1
        reversed_num = int(str(abs(x))[::-1]) * sign
        
        # Check 32-bit signed integer bounds
        return reversed_num if -2**31 <= reversed_num <= 2**31 - 1 else 0

File: 0007-reverse-integer/0007-reverse-integer.go
import (
    "math"
)

func reverse(x int) int {
    sign := 1
    if x < 0 {
        sign = -1
        x = -x
    }

    reversed := 0
    for x > 0 {
        digit := x % 10
        
        // Check for overflow before multiplying
        if reversed > math.MaxInt32/10 {
            return 0
        }

        reversed = reversed*10 + digit
        x /= 10
    }

    result := reversed * sign

    // Check if result is within 32-bit signed integer range
    if result < math.MinInt32 || result > math.MaxInt32 {
        return 0
    }

    return result
}